sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -0,0 +1,186 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { SystemUsersService } from "@sims/services";
+import { CASSupplier, SupplierAddress, SupplierStatus } from "@sims/sims-db";
+import { ProcessSummary } from "@sims/utilities/logger";
+import {
+  CASAuthDetails,
+  CASSupplierResponseItem,
+} from "../../processors/schedulers/cas-integration/models/cas-supplier-response.dto";
+import { Not, Repository, UpdateResult } from "typeorm";
+import { CASIntegrationConfig, ConfigService } from "@sims/utilities/config";
+import { CASService } from "./cas.service";
+
+const CAS_SUPPLIER_ADDRESS_ACTIVE_STATUS = "ACTIVE";
+
+@Injectable()
+export class CASSupplierIntegrationService {
+  private readonly casIntegrationConfig: CASIntegrationConfig;
+  constructor(
+    config: ConfigService,
+    private readonly casService: CASService,
+    private readonly systemUsersService: SystemUsersService,
+    @InjectRepository(CASSupplier)
+    private readonly casSupplierRepo: Repository<CASSupplier>,
+  ) {
+    this.casIntegrationConfig = config.casIntegration;
+  }
+
+  /**
+   * CAS integration process.
+   * Logs on CAS supplier API and request the supplier information for the students with pending supplier information.
+   * @param parentProcessSummary parent process summary.
+   * @param casSuppliers pending CAS suppliers.
+   * @returns a number of update records.
+   */
+  async executeCASIntegrationProcess(
+    parentProcessSummary: ProcessSummary,
+    casSuppliers: CASSupplier[],
+  ): Promise<number> {
+    let suppliersUpdated = 0;
+    const summary = new ProcessSummary();
+    parentProcessSummary.children(summary);
+    try {
+      summary.info("Logging on CAS...");
+      const auth = await this.casService.casLogon();
+      if (auth.access_token) {
+        summary.info("Logon successful.");
+        suppliersUpdated = await this.requestCASAndUpdateSuppliers(
+          casSuppliers,
+          summary,
+          auth,
+        );
+      } else {
+        summary.info("Could not authenticate on CAS.");
+      }
+    } catch (error: unknown) {
+      summary.error("Unexpected error.", error);
+    }
+    return suppliersUpdated;
+  }
+
+  /**
+   * For each pending CAS supplier, request supplier information to CAS API and update local table.
+   * @param casSuppliers pending CAS suppliers.
+   * @param summary log summary.
+   * @param auth CAS auth details.
+   * @returns a number of update records.
+   */
+  private async requestCASAndUpdateSuppliers(
+    casSuppliers: CASSupplier[],
+    summary: ProcessSummary,
+    auth: CASAuthDetails,
+  ) {
+    let suppliersUpdated = 0;
+    for (const casSupplier of casSuppliers) {
```

Comment:
Just double-checking the team's understanding on this, we will not be performing parallel requests for CAS because the amount of expected daily requests would not be enough to justify it. Does that sound correct?